applications/FluidDynamicsApplication/custom_processes/distance_modification_process.cpp

```diff
@@ -272,69 +274,176 @@ void DistanceModificationProcess::ModifyDistance() {
     this->SetContinuousDistanceToSplitFlag();
 }
 
-void DistanceModificationProcess::ModifyDiscontinuousDistance(){
-
+void DistanceModificationProcess::ModifyDiscontinuousDistance()
+{
     auto r_elems = mrModelPart.Elements();
     auto elems_begin = mrModelPart.ElementsBegin();
     const auto n_elems = mrModelPart.NumberOfElements();
+    const size_t n_edges_extrapolated = elems_begin->GetValue(ELEMENTAL_EDGE_DISTANCES_EXTRAPOLATED).size();
 
     // Distance modification
     if (mRecoverOriginalDistance == false) {
-        // Case in where the original distance does not need to be preserved (e.g. CFD)
-        #pragma omp parallel for
-        for (int i_elem = 0; i_elem < static_cast<int>(n_elems); ++i_elem){
-            auto it_elem = elems_begin + i_elem;
+        // Case in which the original distance does not need to be preserved (e.g. CFD)
 
-            // Compute the distance tolerance
-            const double tol_d = mDistanceThreshold*(it_elem->GetGeometry()).Length();
+        // If user has provided flag to calculate elemental edge distances of extrapolated geometry,
+        // modify them according to elemental distances in order to stay consistent with element splitting
+        if (n_edges_extrapolated > 0) {
+            #pragma omp parallel for
+            for (int i_elem = 0; i_elem < static_cast<int>(n_elems); ++i_elem) {
+                auto it_elem = elems_begin + i_elem;
 
-            // Check if the distance values are close to zero
-            Vector &r_elem_dist = it_elem->GetValue(ELEMENTAL_DISTANCES);
-            for (unsigned int i_node = 0; i_node < r_elem_dist.size(); ++i_node){
-                if (std::abs(r_elem_dist(i_node)) < tol_d){
-                    r_elem_dist(i_node) = r_elem_dist(i_node) > 0.0 ? tol_d : -tol_d;
+                // Compute the distance tolerance
+                const double tol_d = mDistanceThreshold*(it_elem->GetGeometry()).Length();
```

Comment:
This is the tolerance we were discussing the other day right?

Reply:
Yes. That was when I still thought that I could use a distance threshold for the extrapolated ratios aswell, before I realized that in order to be consistent, the extrapolated ratios need to be recalculated using the modified elemental distances.